README.md

```diff
@@ -1,18 +1,155 @@
-# ECC Python Cookiecutter Template
+# Python Cookiecutter Template
 
 This is a Cookiecutter template for creating a Python project with Poetry,
 including configurations for various project settings.
 
 ## Usage
 
-1. **Install Cookiecutter** (if you haven't already):
+1. **Install cruft** (if you haven't already):
 
 ```sh
-pip install cookiecutter
+pip install cruft
 ```
 
 2. **Generate a New Project**:
 
 ```sh
-cookiecutter gh:elixir-cloud-aai/cookiecutter-python
+cruft create https://github.com/elixir-cloud-aai/cookiecutter-python.git
 ```
+
+## Configurations
+
+### Package manager - Poetry
+
+This project uses [Poetry](https://python-poetry.org/) as a package manager.
+Check out the commands at the
+[official documentation](https://python-poetry.org/docs/cli/).
+
+### Linters and formatters - Ruff
+
+To lint and format `Python` code files, it uses
+[Ruff](https://docs.astral.sh/ruff), the default configuration is set in the
+`pyproject.toml` file.
+
+```toml
+select = [
+  "B", # flake8-bugbear
+  "E", # pycodestyle
+  "F", # Pyflakes
+  "I", # isort
+  "PL", # pylint
+  "SIM", # flake8-simplify
+  "UP", # pyupgrade
+]
+```
+
+To configure it to your needs, refer to the
+[rules documentation](https://docs.astral.sh/ruff/rules/), and for formatter
+configuration, refer to the
+[configuration documentation](https://docs.astral.sh/ruff/formatter)
+
+### Spell checker - typos
+
+If you want to ignore certain words, add them to the `pyproject.toml` file,
+under the `tool.typos.default.extend-words` key.
+
+```toml
+[tool.typos.default.extend-words]
+mke = 'mke'
+```
+
+For further configuration, refer to the [typos docs](https://pypi.org/project/typos/).
+
+### Static type checker - mypy
+
+Change the configuration in `mypy.ini` file, for further information refer to
+the [documentation](https://mypy.readthedocs.io/en/stable/config_file.html).
+
+### Documentation - Sphinx
+
+Configuration for Sphinx is in the `docs/source/conf.py` file, for further info refer
+to the [Sphinx documentation](https://www.sphinx-doc.org/en/master/usage/configuration.html).
+The default configuration uses
+[furo theme](https://pradyunsg.me/furo/quickstart/) and
+[ReadTheDocs](https://readthedocs.org/) to host the documentation, you can change
+the configuration using `/docs/source/conf.py` and `.readthedocs.yml` file.
+
+### Testing - pytest and pytest-cov
+
+Generate a coverage report using `pytest-cov` and uploads it to codecov.io.
+
+## CI/CD - GitHub Actions
+
+Here are the GitHub Actions Secrets that need to be included in the repository
+settings:
+
+- `PYPI_PASSWORD`: The password for the PyPI account.
+- `CODECOV_TOKEN`: The token for the codecov.io account.
+- `AUTO_UPDATE_GITHUB_TOKEN`: The token for the create a PR to update the template.
```

Comment:
Just thinking of this (not related to this particular line): We should consider configuring [`python-semantic-release`](https://python-semantic-release.readthedocs.io/en/latest/) for PRs against the default branch, as soon as the software reaches 1.0.0.

This will really smoothen and harmonize our releases.

Reply:
I was thinking to push that in this PR as well, for that we will need to genrate `CHANGELOG.md` and for that we will need [commitzen](https://pypi.org/project/commitizen/), our workflow consist of squashing commit and IDK how that will work there, because we have this in commitzen in cloud comp and IMO its just becomes a headache to commit something because we have to go throught a whole form just to write a commit :sob: .

Lemme know if there is a better way or I am missing something, maybe open an issue with your propoosed solution rather than resolving in this PR.

Reply:
No, it will work just fine with semantic commit messages - which we enforce through our PR titles.

Reply:
Anyway, no need to do this now, but we can add it in the future.

Reply:
I mean, I've used this before with the package that I linked, _not_ `commitizen`. There were no problems.

Reply:
gut :+1::